VestingSchedule.sol

```diff
@@ -0,0 +1,227 @@
+    // solhint-disable not-rely-on-time
+
+    modifier onlyRevoker() {
+      require(msg.sender == _revoker, "sender must be the vesting revoker");
+      _;
+    }
+
+    modifier onlyBeneficiary() {
+      require(msg.sender == _revoker, "sender must be the vesting beneficiary");
+      _;
+    }
+
+    using SafeMath for uint256;
+
+    event VestingWithdrawn(address beneficiary, uint256 amount, uint256 timestamp);
+    event VestingRevoked(address revoker, address refundDestination, uint256 refundDestinationAmount, uint256 timestamp);
+
+    // amount that is to be vested
+    uint256 private _vestingAmount;
+
+    // beneficiary of the amount
+    address private _beneficiary;
+
+    // durations in secs. of the entire vesting
+    uint256 private _duration;
+
+    // timestamps for start and cliff starting points. Timestamps are expressed in UNIX time, the same units as block.timestamp.
+    uint256 private _cliff;
+    uint256 private _start;
+
+    // indicates if the contract is revokable
+    bool private _revocable;
+
+    // revoking address and destination
+    address private _refundDestination;
+    address private _revoker;
+
+    // indicates how much of the vested amount has been released for withdrawal
+    uint256 private _released;
+
+    // indicates if the vesting has been revoked. false by default
+    bool _revoked;
+
+    // the time at which the revocation has taken place
+    uint256 _revokeTime;
+
+    /**
+     * @notice A constructor for initialising a new instance of a Vesting Schedule contract
+     * @param beneficiary address of the beneficiary to whom vested tokens are transferred
+     * @param vestingAmount the amount that is to be vested by the contract
+     * @param vestingCliff duration in seconds of the cliff in which tokens will begin to vest
+     * @param vestingStartTime the time (as Unix time) at which point vesting starts
+     * @param vestingPeriodSec duration in seconds of the period in which the tokens will vest
+     * @param revocable whether the vesting is revocable or not
+     * @param revoker address of the person revoking the vesting
+     * @param refundDestination address of the refund receiver after the vesting is deemed revoked
+     * @param vestingContractOwner the owner that of the vesting contract
+     */
+    constructor (address beneficiary,
+                uint256 vestingAmount,
+                uint256 vestingCliff,
+                uint256 vestingStartTime,
+                uint256 vestingPeriodSec,
+                bool    revocable,
+                address revoker,
+                address refundDestination,               
+                address vestingContractOwner) public {
+
+        // do some basic checks
+        require(vestingAmount > 0, "Amount must be positive");
+        require(beneficiary != address(0), "Beneficiary is the zero address");
+        require(revoker != address(0), "Revoker is the zero address");
```

Comment:
Are we sure that the revoker cant be 0. If there is non-revokable contract the revoker should be 0

Reply:
agree, changed